src/devices/Button/GpioButton.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+
+namespace Iot.Device.Button
+{
+    /// <summary>
+    /// GPIO implementation of Button.
+    /// Inherits from ButtonBase.
+    /// </summary>
+    public class GpioButton : ButtonBase
+    {
+        private GpioController _gpioController;
+        private int _buttonPin;
+        private bool _pullUp;
+        private bool _shouldDispose;
+
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Initialization of the button.
+        /// </summary>
+        /// <param name="buttonPin">GPIO pin of the button.</param>
+        /// <param name="pullUp">If the system is pullup (false = pulldown).</param>
+        /// <param name="gpio">Gpio Controller.</param>
+        /// <param name="shouldDispose">True to dispose the GpioController.</param>
+        public GpioButton(int buttonPin, bool pullUp = true, GpioController? gpio = null, bool shouldDispose = true)
+            : this(buttonPin, pullUp, TimeSpan.FromTicks(DefaultDoublePressTicks), TimeSpan.FromMilliseconds(DefaultHoldingMilliseconds), gpio, shouldDispose)
+        {
+        }
+
+        /// <summary>
+        /// Initialization of the button.
+        /// </summary>
+        /// <param name="buttonPin">GPIO pin of the button.</param>
+        /// <param name="pullUp">If the system is pullup (false = pulldown).</param>
+        /// <param name="doublePress">Max ticks between button presses to count as doublepress.</param>
+        /// <param name="holding">Min ms a button is pressed to count as holding.</param>
+        /// <param name="gpio">Gpio Controller.</param>
+        /// <param name="shouldDispose">True to dispose the GpioController.</param>
+        public GpioButton(int buttonPin, bool pullUp, TimeSpan doublePress, TimeSpan holding, GpioController? gpio = null, bool shouldDispose = true)
```

Comment:
I think `pullUp` should be an enum, because we might want to support internal pull-ups/pull-downs as well.